lib/rouge/lexers/kotlin.rb

```diff
@@ -65,6 +65,9 @@ class Kotlin < RegexLexer
         rule %r'/[*].*[*]/', Comment::Multiline # single line block comment
         rule %r'/[*].*', Comment::Multiline, :comment # multiline block comment
         rule %r'\n', Text
+        rule %r'(::)(class)' do
+          groups Operator, Name
+        end
```

Comment:
The following code is in [the visual sample]:

https://github.com/rouge-ruby/rouge/blob/1bd8eb10b9dc914d570fdb487dddf9800f5cf4bd/spec/visual/samples/kotlin#L169

The new rule changes the way this is highlighted. Is this use of `::class` the same as the one you're addressing or does it have a different meaning in Kotlin?

Reply:
First, why it is highlighted differently. I looked into this example in debug mode:

__Before__: When the lexer arrives at `class`:
```
stack: [:root]
...
    got: "class"
    yielding: Keyword, "class"
```

__After__: When the lexer arrives at `class`:
```
stack: [:root]
...
    yielding: Operator, "::"
    yielding: Name, "class"
```

So, my fix prevents this `class` to be interpreted as keyword; however, in your example above no `:class` is pushed onto the stack.

Now about the meaning: The ::-operator creates a reference and can be called with e.g. properties or functions. To answer your question: it's the same meaning. It now depends how you want this to be displayed. I just checked with IntelliJ, here the "things" after the ::-operator are rendered differently depending on what they are. see below:

![Screenshot 2020-09-08 at 08 09 21](https://user-images.githubusercontent.com/118812/92439405-1313fc80-f1ab-11ea-8188-c37a82ac96c8.png)

Reply:
PS: I am fine with both variants (rendering it as the class keyword or rendering it as a name), as long as the superfluous `:class` is not pushed onto the stack :)

Reply:
@contradictioned I think it makes sense to use the `Keyword` token here as well so I've updated the PR to do that. Looks ready to merge to me!